consensus/polybft/state_sync_relayer.go

```diff
@@ -0,0 +1,344 @@
+package polybft
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/umbracle/ethgo"
+)
+
+const (
+	// defaultMaxBlocksToWaitForResend specifies how many blocks should be wait
+	// in order to try again to send transaction
+	defaultMaxBlocksToWaitForResend = uint64(30)
+	// defaultMaxAttemptsToSend specifies how many sending retries for one transaction
+	defaultMaxAttemptsToSend = 6
+)
+
+var (
+	errFailedToExecuteStateSync = errors.New("failed to execute state sync")
+)
+
+// StateSyncRelayer is an interface that defines functions for state sync relayer
+type StateSyncRelayer interface {
+	PostBlock(req *PostBlockRequest) error
+	Init() error
+	Close()
+}
+
+// stateSyncProofRetriever is an interface that exposes function for retrieving state sync proof
+type stateSyncProofRetriever interface {
+	GetStateSyncProof(stateSyncID uint64) (types.Proof, error)
+}
+
+var _ StateSyncRelayer = (*dummyStakeSyncRelayer)(nil)
+
+// dummyStakeSyncRelayer is a dummy implementation of a StateSyncRelayer
+type dummyStakeSyncRelayer struct{}
+
+func (d *dummyStakeSyncRelayer) PostBlock(req *PostBlockRequest) error { return nil }
+
+func (d *dummyStakeSyncRelayer) Init() error { return nil }
+func (d *dummyStakeSyncRelayer) Close()      {}
+
+var _ StateSyncRelayer = (*stateSyncRelayerImpl)(nil)
+
+// StateSyncRelayerStateData is used to keep state of state sync relayer
+type StateSyncRelayerStateData struct {
```

Comment:
Since we only have one field in this struct, I would remove it completely, and just use the `uint64`.